static/src/javascripts/projects/common/modules/analytics/coreVitals.ts

```diff
@@ -0,0 +1,133 @@
+import { getCookie } from '@guardian/libs';
+import { getCLS, getFCP, getFID, getLCP, getTTFB } from 'web-vitals';
+
+export const init = () => {
+	coreVitals();
+};
+
+type CoreWebVitalsPayload = {
+	page_view_id: string | null;
+	received_timestamp: string | null;
+	browser_id: string | null;
+	received_date: string;
+	fid: null | number;
+	cls: null | number;
+	lcp: null | number;
+	fcp: null | number;
+	ttfb: null | number;
+};
+
+const timestamp = new Date();
+const date = new Date().toISOString().slice(0, 10);
+
+const jsonData: CoreWebVitalsPayload = {
+	browser_id: null,
+	page_view_id: null,
+	received_timestamp: timestamp.toISOString(),
+	received_date: date,
+	fid: null,
+	cls: null,
+	lcp: null,
+	fcp: null,
+	ttfb: null,
+};
+
+const coreVitals = (): void => {
+	type CoreVitalsArgs = {
+		name: string;
+		value: number;
+	};
+
+	// Needed when we start using version 2
+	// let hasFCPBeenSent = false;
+
+	const nineDigitPrecision = (value: number) => {
+		// The math functions are to make sure the length of number is <= 9
+		return Math.round(value * 1_000_000) / 1_000_000;
+	};
+
+	const jsonToSend = ({ name, value }: CoreVitalsArgs): void => {
+		switch (name) {
+			case 'FCP':
+				jsonData.fcp = nineDigitPrecision(value);
+				break;
+			case 'CLS':
+				jsonData.cls = nineDigitPrecision(value);
+				break;
+			case 'LCP':
+				jsonData.lcp = nineDigitPrecision(value);
+				break;
+			case 'FID':
+				jsonData.fid = nineDigitPrecision(value);
+				break;
+			case 'TTFB':
+				jsonData.ttfb = nineDigitPrecision(value);
+				break;
+		}
+
+		if (window.guardian.ophan) {
+			jsonData.page_view_id = window.guardian.ophan.pageViewId;
+			// jsonData.browser_id = window.guardian.config.ophan.browserId;
```

Comment:
I don't fully remember why we took this out on DCR, but it feels a bit more maintainable to use this than directly check the cookie, but only if we can.

Reply:
I get this error for that commented line `TS2339: Property 'config' does not exist on type '{ ophan?: { viewId: string; pageViewId: string; } | undefined; }'.`

but we do use `jsonData.browser_id = window.guardian.config.ophan.browserId;` in dcr